viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -647,10 +647,31 @@ void SLAMServiceImpl::SaveMapWithTimestamp() {
     }
 }
 
+void SLAMServiceImpl::AddSensorReading(
+    cartographer::sensor::TimedPointCloudData measurement) {
+    cartographer::transform::Rigid3d tmp_global_pose;
+    bool update_latest_global_pose = false;
+
+    {
+        std::lock_guard<std::mutex> lk(map_builder_mutex);
+        if (measurement.ranges.size() > 0) {
+            trajectory_builder->AddSensorData(kRangeSensorId.id, measurement);
+            auto local_poses = map_builder.GetLocalSlamResultPoses();
+            if (local_poses.size() > 0) {
+                update_latest_global_pose = true;
+                tmp_global_pose = map_builder.GetGlobalPose(trajectory_id,
```

Comment:
What is this trajectory_id? What sets it? Why do we need to provide it to get the global pose?

Reply:
Thanks for looking into this @kim-mishra ! I still have some more questions @kkufieta I see you write the map builder code, so I'm wondering if you might be able to help answer some of them (no worries if not, I know this code was written a while ago):
1. Why is it safe to only use a single trajectory_id for the entire mapping / updating / localizing session?
2. Under what conditions will `local_poses.size()` be 0?
3. What is `kRangeSensorId`?

Reply:
1. In our case, we're starting a new trajectory each time we start up cartographer, and we only need to keep track of that one trajectory (together with the associated `trajectory_id`. 
2. It can happen in the beginning when starting up cartographer. We observed that just because we added a sensor reading, `local_poses` did not necessarily have an entry yet.
3. `kRangeSensorId` defines the sensor type: https://github.com/cartographer-project/cartographer/blob/ef00de2317dcf7895b09f18cc4d87f8b533a019b/cartographer/mapping/trajectory_builder_interface.h#L66-L73